### DISCUSSION TITLE
How do I directly link my third-party React Native library, which includes native code, to the host application?

### DISCUSSION THREAD

**Discussion by chiragramimi** [callstack/react-native-builder-bob#626] [General]
How can I directly link my third-party React Native library, which includes native code, to the host application?

For example, I have added the following packages to my library module:

```
  "@react-native-voice/voice": "^3.2.4",
    "react-native-image-picker": "^7.1.2",
    "@notifee/react-native": "^7.8.2",
    "react-native-pdf": "^6.7.4",
    "lottie-ios": "4.3.4",
    "lottie-react-native": "6.4.1"

```

However, the iOS pods for these libraries are not being directly installed in the host application, even though I have included them in peerDependencies and devDependencies.

**Comment by chiragramimi** [comment]
I already refer this docummentaiton

https://callstack.github.io/react-native-builder-bob/faq#how-do-i-add-a-react-native-library-containing-native-code-as-a-dependency-in-my-library

**Comment by satya164** [comment]
> the iOS pods for these libraries are not being directly installed in the host application

What does "not being directly installed" mean?

If a third-party library is not being auto-linked then it's up to whoever made the library to fix that. It's unrelated to your library setup.

**Reply by chiragramimi** [reply]
Thanks, @satya164, for your quick response! Could you please provide the steps for auto-linking the library? It would be really helpful for me

**Reply by satya164** [reply]
"auto"linking is automatic. There are no steps other than installing the package. If it's not working for a third party library then you need to open an issue for that library.
